test_scripts/Defects/7_1/3580_Revoked_app_activated_via_device_consent.lua

```diff
@@ -0,0 +1,64 @@
+---------------------------------------------------------------------------------------------------
+-- User story: https://github.com/SmartDeviceLink/sdl_core/issues/3580
+--
+-- Description:
+-- HMI responds with UNSUPPORTED_RESOURCE to Speak component of SubtleAlert
+--
+-- Preconditions:
+-- 1) Clean environment
+-- 2) SDL, HMI, Mobile session started
+-- 3) Registered app
+--
+-- Steps: 
+-- 1) Activate app with consent
+--
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local json = require("modules/json")
+local runner = require('user_modules/script_runner')
+local common = require("user_modules/sequences/actions")
+local utils = require ('user_modules/utils')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+runner.testSettings.restrictions.sdlBuildOptions = { { extendedPolicy = { "EXTERNAL_PROPRIETARY" } } }
+
+--[[ Local Functions ]]
+local function PTUfunc(tbl)
+    tbl.policy_table.app_policies[config.application1.registerAppInterfaceParams.fullAppID] = json.null
+end
+
+function activateApp(pAppId)
+    if not pAppId then pAppId = 1 end
+    local requestId = common.hmi.getConnection():SendRequest("SDL.ActivateApp", { appID = common.app.getHMIId(pAppId) })
+    common.hmi.getConnection():ExpectResponse(requestId, { result = {
+        code = 4,
+        isAppRevoked = true,
+        method = "SDL.ActivateApp"
+    }})
+  :Do(function(_,_)
+      local RequestId1 = common.hmi.getConnection():SendRequest("SDL.GetUserFriendlyMessage", {language = "EN-US", messageCodes = {"DataConsent"}})
+      local RequestId2 = common.hmi.getConnection():SendRequest("SDL.GetUserFriendlyMessage", {language = "EN-US", messageCodes = {"AppUnsupported"}})
+
+      EXPECT_HMIRESPONSE(RequestId1,{result = {code = 0, method = "SDL.GetUserFriendlyMessage"}})
+      :Do(function(_,_)
+        common.hmi.getConnection():SendNotification("SDL.OnAllowSDLFunctionality", {allowed = true, source = "GUI", device = {id = utils.getDeviceMAC(), name = utils.getDeviceName()}})
+        end)
+
+      EXPECT_HMIRESPONSE(RequestId2,{result = {code = 0, method = "SDL.GetUserFriendlyMessage"}})
+        
```

Comment:
I think you would need to add another expectation here to match up with the issue (currently this test passes on develop):
```
      common.getHMIConnection():ExpectRequest("BasicCommunication.ActivateApp"):Times(0)
```

Reply:
Added in https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2505/commits/7ffb9e92e9d995f843068e807a2f61437c5db45b